packages/eslint-plugin/src/rules/naming-convention.ts

```diff
@@ -394,20 +394,21 @@ export default util.createRule<Options, MessageIds>({
           },
         },
 
-      'TSPropertySignature[computed = false]': {
-        validator: validators.typeProperty,
-        handler: (
-          node: TSESTree.TSPropertySignatureNonComputedName,
-          validator,
-        ): void => {
-          const modifiers = new Set<Modifiers>([Modifiers.public]);
-          if (node.readonly) {
-            modifiers.add(Modifiers.readonly);
-          }
+      'TSPropertySignature[computed = false][typeAnnotation.typeAnnotation.type != "TSFunctionType"]':
+        {
+          validator: validators.typeProperty,
+          handler: (
+            node: TSESTree.TSPropertySignatureNonComputedName,
+            validator,
+          ): void => {
+            const modifiers = new Set<Modifiers>([Modifiers.public]);
+            if (node.readonly) {
+              modifiers.add(Modifiers.readonly);
```

Comment:
Oh sorry @kozlovvski - the codecov app had a delay today and I missed this. Could you please add test cases to exercise this line? It's not covered right now.

Reply:
@JoshuaKGoldberg I can try, but I am not sure what this part does. I didn't adjust the `handler` -> it's marked as changed due to indentation

Reply:
@JoshuaKGoldberg okay, I think it should be fixed now 🙌🏻